src/jwt.rs

```diff
@@ -12,7 +12,10 @@ pub fn validate_token_rs256<CustomClaims: Serialize + DeserializeOwned>(
     // Peek at the token metadata before verification and retrieve the key identifier,
     // in order to pick the right key out of the JWK set.
     let metadata = Token::decode_metadata(&token_string)?;
-    let key_id = metadata.key_id().unwrap();
+    let key_id = match metadata.key_id() {
+        None => return Err(Error::msg("decode_metadata returned None")),
```

Comment:
I'd like to suggest a different error message here to avoid confusion – [`decode_metadata`](https://docs.rs/jwt-simple/0.9.4/jwt_simple/token/struct.Token.html#method.decode_metadata) above returns `Result<TokenMetadata, Error>`.

```suggestion
        None => return Err(Error::msg("Failed to decode public key identifier for token")),
```